leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/LeshanClientDemo.java

```diff
@@ -398,6 +372,10 @@ public static void createAndStartClient(String endpoint, String localAddress, in
                 initializer.setInstancesForObject(SECURITY, x509(serverURI, 123, clientCertificate.getEncoded(),
                         clientPrivateKey.getEncoded(), serverCertificate.getEncoded()));
                 initializer.setInstancesForObject(SERVER, new Server(123, 30, BindingMode.U, false));
+            } else if (useOSCore) {
+                initializer.setInstancesForObject(SECURITY, secOSCore(serverURI, 123));
+                initializer.setInstancesForObject(OSCORE, Oscore.set("11223344", "AA", "BB")); //Hardcoded values
+                initializer.setInstancesForObject(SERVER, new Server(123, 30, BindingMode.U, false));
```

Comment:
Regarding your remarks here : 
> One issue I had was how exactly to link the Security and Oscore objects together. I now see your suggestion here: #726 (comment)

To be able to do that, you will probably need to add a new XML version of [security object](http://www.openmobilealliance.org/release/LightweightM2M/V1_1_1-20190617-A/HTML-Version/OMA-TS-LightweightM2M_Core-V1_1_1-20190617-A.html#Table-E1-1-LwM2M-Object-LWM2M-Security-object-definition).

You should get it [here](http://www.openmobilealliance.org/wp/OMNA/LwM2M/LwM2MRegistry.html)

Reply:
The security object xml has been updated and a link to the OSCORE object introduced in commit ecebdfa09ce8028fa02d0d26c448ce3029069479